### PR TITLE
docs and glibc: explicit comments on glibc executables not being in the first output

### PR DESCRIPTION
**Pull request by 1pakch**
###### Motivation for this change
The chapter on multiple outputs mentions the commonly adopted convention to put the binaries in the package's first output. The `glibc` package contradticts this convention and I have spent quite some time figuring out why I cannot find `ldd` in `glibc` as prescribed. 

This commit adds the brief description of this deliberate exception in `glibc` package and why it exists (#30984) to the nixpkgs guide 

###### Things done
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions

---

### CODE REVIEW

**Comment by ilya-kolpakov** [comment]
Added comments to `glibc` package as well.

File: doc/multiple-output.xml [hunk 1]
```diff
@@ -36,10 +36,16 @@
...
   <programlisting>outputs = [ "bin" "dev" "out" "doc" ];</programlisting>
-  <para>Often such a single line is enough.  For each output an equally named environment variable is passed to the builder and contains the path in nix store for that output.  By convention, the first output should contain the executable programs provided by the package as that output is used by Nix in string conversions, allowing references to binaries like <literal>${pkgs.perl}/bin/perl</literal> to always work. Typically you also want to have the main <varname>out</varname> output, as it catches any files that didn't get elsewhere.</para>
-
+  <para>Often such a single line is enough.  For each output an equally named environment variable is passed to the builder and contains the path in nix store for that output. Typically you also want to have the main <varname>out</varname> output, as it catches any files that didn't get elsewhere.</para>
```
Comment (FRidh) [inline doc/multiple-output.xml]: Do not use "you"

File: doc/multiple-output.xml [hunk 2]
```diff
@@ -36,10 +36,16 @@
...
+    <para>The <literal>glibc</literal> package is a deliberate single exception to the <quote>binaries first</quote> convention. The <literal>glibc</literal> has <literal>libs</literal> as its first output allowing the libraries provided by <literal>glibc</literal> to be referenced directly (e.g. <literal>${stdenv.glibc}/lib/ld-linux-x86-64.so.2</literal>). The executables provided by <literal>glibc</literal> can be accessed via its <literal>bin</literal> attribute (e.g. <literal>${stdenv.glibc.bin}/bin/ldd</literal>).</para>
+    <para>The reason for why <literal>glibc</literal> deviates from the convention is because referencing a library provided by <literal>glibc</literal> is a very common operation among Nix packages. For instance, third-party executables packaged by Nix are typically patched and relinked with the relevant version of <literal>glibc</literal> libraries from Nix packages (please see the documentation on <link xlink:href="https://nixos.org/patchelf.html">patchelf</link> for more details).</para>
```
Comment (FRidh) [inline doc/multiple-output.xml]: The motivation for

File: doc/multiple-output.xml (hunk 1 above)
Comment (ilya-kolpakov) [reply doc/multiple-output.xml]: I merely moved it this sentence. Btw, Is there a style guide. If there is, it is not that easily discoverable IMO.

File: doc/multiple-output.xml (hunk 2 above)
Comment (ilya-kolpakov) [reply doc/multiple-output.xml]: Maybe the section id needs to be more in line with the title as well.

**Comment by ilya-kolpakov** [comment]
I understand that modifications of `glibc` trigger massive rebuilds but this one changes only docs and comments so it won't break everything and can be merged if there are no objections IMO.

File: doc/multiple-output.xml (hunk 2 above)
Comment (ilya-kolpakov) [reply doc/multiple-output.xml]: I am not quite sure I understand your comment. Are you asking what is the motivation for including in the docs?

**Comment by matthewbauer** [comment]
This seems helpful to me. @FRidh any objections to merging? We can clean it up a little later.

**Comment by FRidh** [comment]
Nope

**Review by vcunat — APPROVED** [review]

**Pull request merged**